### PR TITLE
v5 Docs followup

### PR DESCRIPTION
**Pull request by mdo**
With #28450 (major forms rewrite) I changed the designed of the docs to get back to how we used to build this thing in v3 and earlier. v4's docs have always felt like a web app to me instead of an documentation site. Three columns of content, most of which stayed stickied was also a tad frustrating for me. More importantly though, I felt things were lacking polish, attention to detail, etc.

This PR will tackle the todos from @MartijnCuppens's comment and from @patrickhlauke's issue at #29047.

### Changes

Here's some clarification and additional fixes (including todos) to follow-up:

- [x] Clarify and update padding changes to `<ol>` and `<ul>` in Reboot. I added `padding-right` originally by mistake thinking I needed to reset that, but really I only needed `padding-left` redone.
- [x] Add new Icons component to the migration docs (I created these myself to replace the Open Iconic ones we'd been using). More icons are planned to come, but for now it focuses on a handful for our components.
- [x] Clarify migration docs for the changes in apperance and behavior for form components (namely checks and files).
- [x] Add note to migration docs about table changes.
- [x] Fix remaining tables that were lacking `.table`.
- [x] Improve icons component docs with code snippets, design decisions, and usage guidelines.
- [ ] ~~Look into `box-shadow`s support on icons with custom form controls~~
- [x] After #29029 merges, add search form to the docs homepage
- [x] Consider moving sidebar icon to `::before` per https://github.com/twbs/bootstrap/pull/29029#issuecomment-511722549 **Done in @XhmikosR's PR**
- [x] Fix color contrast in navbar
- [x] Update icons one more time—add some new ones, rename the class from `.booticon` to `.bi` because it's shorter and comes with fewer jokes, and remove unused styles in docs CSS.
- [x] Make navbar, subnav, and sidebar work on mobile
- [x] Move the icons to includes
- [ ] Move icons outside of content folder
- [ ] Stop inlining the SVGs; we have http/2 so it should be pretty fast anyway
- [ ] Verify if we need to add `role` or `focusable="false"`
- [ ] Move the nav changes to a separate PR

---

Explaining some other decisions...

### Tables

Tables saw a redesign to simplify their CSS and overall appearance. I also took the opportunity to remove several lines of custom docs CSS that re-applied the Bootstrap table styles to Markdown tables. Doing so meant rewriting the tables to allow explicit classes, which I was fine to do given we're inconsistent in our Markdown/HTML tables usage.

See https://github.com/twbs/bootstrap/pull/28450/files#diff-e923be674b22e7e32a5d742c5a05b3ae, https://github.com/twbs/bootstrap/pull/28450/commits/c8d1a0edb5da7bf4c4e4fbe714ec9500e5774c99, and https://github.com/twbs/bootstrap/pull/28450/commits/f9f9e45ec8af2344786853d041f2a8b9bbeb5d20.

### Larger form controls

This was intentional, and really only applies to the checks, radios, and switches. They're larger than the default and inconsistently styled ones each OS sees (macOS ones are quite small) and scale now to `font-size` or explicit dimension changes. I'll look into these not working in IE, but I thought my early tests in there showed no problems.

### Icons

Still working on more here, but again, these are custom icons I've made to replace the ones we were using from external libraries. Goal is to have a toolkit focused set at launch to support our components. Maybe more icons can be added after that. In the future, this could be it's own project I suppose.

### Docs design changes

While the intro to this PR covers much of it, the changes here are meant to simplify our docs and accelerate movement around the docs pages. The sidebar has been overhauled to always have all links available to each docs page vs only the ones in the group you're in (e.g., Components). Stylistically, I'm working to simplify the main navbar for usability and improve the page content design with fewer custom styles.

Preview: <https://deploy-preview-29048--twbs-bootstrap.netlify.com/>

---

@twbs/css-review Keep the questions and feedback coming.

### CODE REVIEW

**Comment by XhmikosR** [comment]
Adding a todo:

- [ ] Move the icons to includes maybe

**Comment by XhmikosR** [comment]
@mdo how about moving all icons to a new folder in partials? This way we could automate their optimization etc.

**Comment by mdo** [comment]
@XhmikosR Just pushed up changes to add all the SVGs (sorry, I know it's the wrong directory) and embed them that way. I'm unsure how Hugo handles things I can loop over from a directory, so I need to revisit that. Any pointers would be great, I'd love to better understand how to use Hugo :). Additionally I'm hoping to add some scripts to handle the SVGs better.

**Comment by XhmikosR** [comment]
I'm gonna rebase and clean the branch and see what I can do later.

**Comment by XhmikosR** [comment]
Is the subnav supposed to be shown in the homepage and everywhere?

I will look into automating the icons inclusion later.

**Comment by XhmikosR** [comment]
@mdo check out the last patch. Don't merge this yet, I really want to make sure the nav changes are OK this time. Ideally they should be split in a separate PR and the same for the iconic removal.

**Comment by XhmikosR** [comment]
Also, the folder you have the icons currently copies the SVGs in the dist folder which I believe is redundant.

**Comment by XhmikosR** [comment]
And I'm unsure if we should add `focusable="false"` or `role="img"`....

**Comment by XhmikosR** [comment]
So, @mdo how about we split the nav changes to a separate PR and most importantly for this PR, move the icons folder somewhere outside of content?

File: site/content/docs/4.3/components/icons.md [hunk 1]
```diff
@@ -10,6 +10,21 @@ toc: true
 
 Meet the Bootstrap icons, a limited set of icons designed first and foremost to work with Bootstrap's components. From form controls to navigation, these icons are the perfect complement to our toolkit. They're SVGs, so they'll scale quickly and easily to any size, they can be easily recolored with CSS, and more. They're also included in Bootstrap under the MIT license.
 
+<div class="bd-bi-grid d-flex flex-wrap mb-5">
```
Comment (XhmikosR) [inline site/content/docs/4.3/components/icons.md]: Need to check if we can reuse the shortcode

File: site/content/docs/4.3/components/icons.md [hunk 2]
```diff
@@ -10,6 +10,21 @@ toc: true
...
+  {{< listIcons.inline >}}
+  {{- $dirName := printf "site/content/docs/%s/icons/" .Site.Params.docs_version -}}
+
+  {{- range (readDir $dirName) -}}
+    {{- $filenameWithExt := split .Name "." -}}
+    {{- $filename := index $filenameWithExt 0 -}}
+    {{- $name := humanize $filename -}}
+    <div class="p-3 mr-2 mb-2 border rounded">
+      {{ readFile (printf "%s%s" $dirName .Name) }}
```
Comment (XhmikosR) [inline site/content/docs/4.3/components/icons.md]: Maybe we should add tooltips with the class names?

**Comment by mdo** [comment]
Split via cherry-pick to #29091 and #29092.

**Pull request closed without merging**